TripPlaner/src/main/java/com/singapore/TripPlaner/Service/PlaceService.java

```diff
@@ -1,37 +1,41 @@
 package com.singapore.TripPlaner.Service;
 
-//
-//import java.util.List;
-//import java.util.function.Function;
-//import java.util.stream.Collectors;
-//
-//public class PlaceService extends Reader {
-//    public void findAllPlacesByCity(int index) {
-//        List<Places> place = getAllPlaces(Places.class);
-//        System.out.println(place.stream().filter(p -> p.getFromCity() == index).collect(Collectors.toList()));
-//    }
-//
-//    public void findAllPlaces() {
-//        List<Persistent> place = getList(Places.class);
-//        for (Persistent listOfPlaces : place) {
-//            Places oParsed = (Places) listOfPlaces;
-//            System.out.println(oParsed);
-//        }
-//    }
-//
-//    public void findAllPlacesByCityOnlyNames(int index) {
-//        List<Places> place = getAllPlaces(Places.class);
-//        List<Places> filteredList = filterListByCityIndex(index, place);
-//        showOnlyNamesCity(filteredList).forEach(System.out::println);
-//
-//    }
-//
-//    private List<Places> filterListByCityIndex(int index, List<Places> place) {
-//        return place.stream().filter(p -> p.getFromCity() == index).collect(Collectors.toList());
-//    }
-//
-//    private List<String> showOnlyNamesCity(List<Places> list) {
-//        Function<Places, String> onlyName = s -> s.getName();
-//        return list.stream().map(onlyName).collect(Collectors.toList());
-//    }
-//}
+
+import com.singapore.TripPlaner.Model.*;
+import com.singapore.TripPlaner.Service.dataacces.Reader;
+import com.singapore.TripPlaner.Service.dataacces.Writer;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+@Component
+public class PlaceService extends PersistentAbstract{
+    private final Reader reader;
+    private final Writer writer;
+
+    public PlaceService(Reader reader, Writer writer) {
+        this.reader = reader;
+        this.writer = writer;
+    }
+
+    public void findAllPlacesByCity(int index) {
```

Comment:
```suggestion
    public List<Place> findAllPlacesByCity(String placeType) {
```
a tu wtedy zamiast void  List Place